clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -145,6 +149,7 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
       }
     } else if (FunctionDecl *FD = dyn_cast<FunctionDecl>(D)) {
       FVCons = Info.getVariable(FD, Context);
+      TFD = FD;
     } else if (DeclaratorDecl *DD = dyn_cast<DeclaratorDecl>(D)) {
       FVCons = Info.getVariable(DD, Context);
```

Comment:
Why not also have `TFD = DD` here? Isn't it possible that this is a function pointer and we'll need to note the bounds on that too?